File: examples/matlab/sections.m
matrix = [1 2 3; 4 5 6];
row = matrix(2, :);
column = matrix(:, 2);
block = matrix(:, 1:2);
linear = matrix(:);
odd = matrix(1, 1:2:3);
disp(sum(row) + sum(column) + numel(block) + linear(2) + sum(odd))

File: examples/fortran/sections.f90
program sections
  implicit none
  integer :: matrix(2, 3) = reshape([1, 4, 2, 5, 3, 6], [2, 3])
  integer :: row(3)
  integer :: column(2)
  integer :: block(2, 2)
  integer :: reverse(3)

  row = matrix(2, :)
  column = matrix(:, 2)
  block = matrix(:, 1:2)
  reverse = matrix(2, 3:1:-1)
  print *, sum(row) + sum(column) + size(block) + reverse(1)
end program sections
